api/src/main/java/net/kyori/adventure/text/format/TextDecoration.java

```diff
@@ -99,6 +99,28 @@
     return new TextDecorationAndStateImpl(this, state);
   }
 
+  /**
+   * An alias for {@link #as(boolean)}.
+   *
+   * @param state the state
+   * @return a {@link TextDecorationAndState}
+   * @since 4.10.0
+   */
+  public final @NotNull TextDecorationAndState withState(final boolean state) {
+    return this.as(State.byBoolean(state));
+  }
+
+  /**
+   * An alias for {@link #as(State)}.
+   *
+   * @param state the state
+   * @return a {@link TextDecorationAndState}
+   * @since 4.10.0
+   */
+  public final @NotNull TextDecorationAndState withState(final @NotNull State state) {
+    return this.as(state);
+  }
```

Comment:
would you be able to add `withState(TriState)` (and `as(TriState)`) overloads here? kinda scope creep but merging #495 sparked the thought

Reply:
Do we need an `as(TriState)` overload? As it's new API we don't need another method here that does the same thing. Although I'm also for deprecating the existing `as` method, so that might change if we decide not to.